File: dart_sample/lib/demos/cupertino/cupertino_icons_demo_page.dart
import 'package:cupertino_ui/cupertino_ui.dart';
import 'package:material_ui/material_ui.dart';

class CupertinoIconsDemoPage extends StatelessWidget {
  const CupertinoIconsDemoPage({super.key});

  static const List<(String, IconData)> _samples = <(String, IconData)>[
    ('heart_fill', CupertinoIcons.heart_fill),
    ('bell_fill', CupertinoIcons.bell_fill),
    ('camera', CupertinoIcons.camera),
    ('person_2_fill', CupertinoIcons.person_2_fill),
    ('map_fill', CupertinoIcons.map_fill),
    ('gear', CupertinoIcons.gear),
    ('search', CupertinoIcons.search),
    ('plus_circle', CupertinoIcons.plus_circle),
    ('chevron_back', CupertinoIcons.chevron_back),
    ('arrow_left_right', CupertinoIcons.arrow_left_right),
    ('waveform_path_ecg', CupertinoIcons.waveform_path_ecg),
    ('videocam_circle_fill', CupertinoIcons.videocam_circle_fill),
  ];

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 12,
      children: <Widget>[
        const Text(
          'Cupertino icons',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'Representative legacy, SF Symbols, directional, alias, and '
          'high-range glyphs.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Wrap(
          spacing: 12,
          runSpacing: 12,
          children: <Widget>[
            for (final (String name, IconData icon) in _samples)
              _buildTile(name, icon),
          ],
        ),
      ],
    );
  }

  static Widget _buildTile(String name, IconData icon) {
    return Container(
      width: 132,
      padding: const EdgeInsets.all(12),
      decoration: BoxDecoration(
        color: const Color(0xFFF2F2F7),
        borderRadius: BorderRadius.circular(12),
      ),
      child: Column(
        spacing: 8,
        children: <Widget>[
          Icon(icon, size: 34, color: const Color(0xFF007AFF)),
          Text(
            name,
            textAlign: TextAlign.center,
            style: const TextStyle(fontSize: 11, color: Colors.black),
          ),
        ],
      ),
    );
  }
}

File: src/Sample/Plumix.Sample/Demos/Cupertino/CupertinoIconsDemoPage.cs
using System.Linq;
using Avalonia;
using Avalonia.Media;
using Plumix.Cupertino;
using Plumix.Rendering;
using Plumix.UI;
using Plumix.Widgets;

// Dart parity source (reference): dart_sample/lib/demos/cupertino/cupertino_icons_demo_page.dart (exact sample parity)

namespace Plumix;

public sealed class CupertinoIconsDemoPage : StatelessWidget
{
    private static readonly (string Name, IconData Icon)[] Samples =
    [
        ("heart_fill", CupertinoIcons.HeartFill),
        ("bell_fill", CupertinoIcons.BellFill),
        ("camera", CupertinoIcons.Camera),
        ("person_2_fill", CupertinoIcons.Person2Fill),
        ("map_fill", CupertinoIcons.MapFill),
        ("gear", CupertinoIcons.Gear),
        ("search", CupertinoIcons.Search),
        ("plus_circle", CupertinoIcons.PlusCircle),
        ("chevron_back", CupertinoIcons.ChevronBack),
        ("arrow_left_right", CupertinoIcons.ArrowLeftRight),
        ("waveform_path_ecg", CupertinoIcons.WaveformPathEcg),
        ("videocam_circle_fill", CupertinoIcons.VideocamCircleFill),
    ];

    public override Widget Build(BuildContext context)
    {
        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 12.0,
            children:
            [
                new Text("Cupertino icons", fontSize: 20.0, color: Colors.Black),
                new Text(
                    "Representative legacy, SF Symbols, directional, alias, and high-range glyphs.",
                    fontSize: 14.0,
                    color: Color.Parse("#8A000000")),
                new Wrap(
                    spacing: 12.0,
                    runSpacing: 12.0,
                    children:
                    [
                        .. Samples.Select(sample => BuildTile(sample.Name, sample.Icon)),
                    ]),
            ]);
    }

    private static Widget BuildTile(string name, IconData icon)
    {
        return new Container(
            width: 132.0,
            padding: new Thickness(12.0),
            decoration: new BoxDecoration(
                Color: Color.Parse("#FFF2F2F7"),
                BorderRadius: BorderRadius.Circular(12.0)),
            child: new Column(
                spacing: 8.0,
                children:
                [
                    new Icon(icon, size: 34.0, color: Color.Parse("#FF007AFF")),
                    new Text(name, fontSize: 11.0, color: Colors.Black, textAlign: TextAlign.Center),
                ]));
    }
}
